### PR TITLE
[server][Action] Reexecute action mechanism.

### PR DESCRIPTION
**Pull request by kz0817**
This is a revised version of #2471
====
This patch adds a feature that executes unfinished actions
when Hatohol server restarts.

Currently, when the Hatohol server craashes or quits forcely,
running and queued actions aren't taken care on the restart of
Hatohol server.

### CODE REVIEW

File: server/src/ActionManager.cc [hunk 1]
```diff
@@ -549,6 +549,52 @@ void ActionManager::checkEvents(const EventInfoList &eventList)
...
+	}
+
+	for (const auto &actionLog: actionLogList) {
+		EventInfoList eventList;
+		EventsQueryOption eventOption(USER_ID_SYSTEM);
+
+		eventOption.setTargetServerId(actionLog.serverId);
+		eventOption.setEventIds({actionLog.eventId});
+		dbMonitoring.getEventInfoList(eventList, eventOption);
+		if (eventList.empty())
```
Comment (kz0817) [inline server/src/ActionManager.cc]: Do we have to show warning in this condition ?

File: server/src/ActionManager.cc [hunk 2]
```diff
@@ -549,6 +549,52 @@ void ActionManager::checkEvents(const EventInfoList &eventList)
...
+		        continue;
+		auto const &eventInfo = *eventList.cbegin();
+
+		ActionDefList actionList;
+		ActionsQueryOption actionOption(USER_ID_SYSTEM);
+		ActionIdList actionIdList{actionLog.actionId};
+		actionOption.setActionIdList(actionIdList);
+		dbAction.getActionList(actionList, actionOption);
+
+		if (actionList.empty())
```
Comment (kz0817) [inline server/src/ActionManager.cc]: Here too.

File: server/src/main.cc [hunk 3]
```diff
@@ -244,6 +245,11 @@ int mainRoutine(int argc, char *argv[])
 
 	// setup configuration database
 	ThreadLocalDBCache cache;
+
+	// Re execute unfinished action
+	ActionManager actionManager;
+	actionManager.reExecuteUnfinishedAction();
```
Comment (cosmo0920) [inline server/src/main.cc]: memo: I've noticed that Hatohol server blocks this line when bunch of unfinished actions are remained.
But for now, this implementation is no problem to re-execute unfinished actions.

File: server/src/ActionManager.cc (hunk 1 above)
Comment (masa0612) [inline server/src/ActionManager.cc]: I think it is better to show logs

File: server/src/ActionManager.cc (hunk 2 above)
Comment (masa0612) [inline server/src/ActionManager.cc]: Here too

**Comment by kz0817** [comment]
I added an error messages.
Please merge the PR if there's no problem.

**Pull request merged**